WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift

```diff
@@ -71,6 +83,27 @@ struct AnalyticsTimeRangeCard: View {
         .padding([.top, .bottom])
         .frame(maxWidth: .infinity)
     }
+
+    /// Tracks the range selection internally to determine if the custom range selection should be presented or not.
+    /// If custom range selection is not needed, the internal selection is forwarded to `selectionType`.
+    ///
+    private func internalSelectionBinding() -> Binding<AnalyticsHubTimeRangeSelection.SelectionType> {
+        .init(
+            get: {
+                selectionType
```

Comment:
This internal selection binding is really clever!

One small issue with this getter: Because the list is populated with `AnalyticsHubTimeRangeSelection.SelectionType.allCases` and the Custom row is defined as `.custom(start: nil, end: nil)`, the Custom row never shows as selected in the list. If you select a custom date range, and then open the date range selector again, nothing is selected:

<img src="https://user-images.githubusercontent.com/8658164/206493964-56f9d5ed-c2bb-42d5-9370-0a42bb1f1cea.png" width="300px">

One solution could be to replace the getter with something like this:

```suggestion
                switch selectionType {
                    // If a `custom` case is set return one with nil values so the Custom row is selected
                case .custom:
                    return .custom(start: nil, end: nil)
                default:
                    return selectionType
                }
```

Reply:
I think this will be solved more elegantly once we decouple the Business options from the UI options.
I'll make sure handle that on https://github.com/woocommerce/woocommerce-ios/issues/8340

Reply:
In the mean time, I've adopted to suggestion as it does not hunt to have it right now!